### PR TITLE
[Model Monitoring] Treat special chars properly in the inference flow 

### PR DESCRIPTION
**Pull request by jond01**
Fix [ML-4635](https://jira.iguazeng.com/browse/ML-4635?focusedCommentId=115430&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-115430)

- Remove the name normalization function duplication
- Extract the `_get_features` function from the model monitoring feature set
  - The name norm takes place here - it's the main logical change of this PR
- Add a system test

### CODE REVIEW

File: server/api/crud/model_monitoring/model_endpoints.py [hunk 1]
```diff
@@ -190,13 +197,49 @@ def patch_model_endpoint(
...
+        return features
+
     @staticmethod
     def create_monitoring_feature_set(
+        features: list[mlrun.feature_store.Feature],
         model_endpoint: mlrun.common.schemas.ModelEndpoint,
-        model_obj: mlrun.artifacts.ModelArtifact,
         db_session: sqlalchemy.orm.Session,
-        run_db: server.api.rundb.sqldb.SQLRunDB,
-    ):
```
Comment (Eyal-Danieli) [inline server/api/crud/model_monitoring/model_endpoints.py:199]: update the docstring of `create_monitoring_feature_set` (remove `model_obj` and `run_db` + add `features`)

File: tests/system/model_monitoring/test_model_monitoring.py [hunk 2]
```diff
@@ -959,3 +962,88 @@ def test_model_monitoring_with_kafka_stream(self):
 
         assert model_endpoint.status.metrics["generic"]["latency_avg_5m"] > 0
         assert model_endpoint.status.metrics["generic"]["predictions_count_5m"] > 0
+
+
+@TestMLRunSystem.skip_test_if_env_not_configured
+@pytest.mark.enterprise
+class TestInferenceWithSpecialChars(TestMLRunSystem):
```
Comment (Eyal-Danieli) [inline tests/system/model_monitoring/test_model_monitoring.py:969]: nice!

**Review by Eyal-Danieli — CHANGES_REQUESTED** [review]
Great!
just a small docstring fix

File: server/api/crud/model_monitoring/model_endpoints.py (hunk 1 above)
Comment (jond01) [reply server/api/crud/model_monitoring/model_endpoints.py:199]: Good catch - fixed: 19d0184

**Review by Eyal-Danieli — APPROVED** [review]

**Pull request merged**